src/provisioner/validators.py

```diff
@@ -0,0 +1,199 @@
+#!/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import errno
+import inspect
+import argparse
+
+from cortx.utils.conf_store import Conf
+from cortx.provisioner.error import CortxProvisionerError
+from cortx.provisioner.config_store import ConfigStore
+
+
+class Validator:
+    """ Validator Framework """
+
+    name = "validator"
+
+    def process(*args):
+        """ This interface needs to be implemented in derived class """
+        pass
+
+    @staticmethod
+    def validate(validations: list, *args):
+        """ Return the validator after parsing the validator list """
+
+        validators = [
+            y for x, y in inspect.getmembers(sys.modules[__name__])
+            if x.endswith('Validator') and y.name in validations]
+        for validator in validators:
+            validator.process(args)
+
+
+class ConfigValidator(Validator):
+    """ Config Validator """
+
+    name = "Config"
+    _cortx_conf_url = "yaml:///etc/cortx/cluster.conf"
+    _solution_index = "solution_config"
+
+    @staticmethod
+    def process(*args):
+        """ Process input parameters """
+
+        parser = argparse.ArgumentParser('provisioner_config_validator')
+        parser.add_argument(
+            '-f', dest='solution_conf', help='Solution Config URL')
+        parser.add_argument(
+            '-c', dest='cortx_conf', nargs='?', help='CORTX Config URL')
+        input_args = parser.parse_args()
+
+        if input_args.solution_conf is None:
+            raise CortxProvisionerError(
+                errno.EINVAL, 'Please provide solution_conf url')
+
+        solution_conf_url = input_args.solution_conf
+        cortx_conf_url = input_args.cortx_conf
+        if cortx_conf_url is None:
+            cortx_conf_url = ConfigValidator._cortx_conf_url
+
+        ConfigValidator().validate(solution_conf_url, cortx_conf_url)
+
+    def validate(self, solution_conf_url, cortx_conf_url):
```

Comment:
Only process() needs to be implemented in this class. 
Please rename all the other functions to be starting with "_", e.g. _check